src/systemcmds/tests/test_IntrusiveQueue.cpp

```diff
@@ -90,8 +92,20 @@ bool IntrusiveQueueTest::test_push()
 	// pop all elements
 	for (int i = 0; i < 100; i++) {
 		auto node = q1.front();
+
+		if (i < 100 - 1) {
+			ut_assert_false(q1.empty());
+		}
+
 		q1.pop();
```

Comment:
You could check that this is equal to `node`.

Reply:
Done.